sretoolbox/container/image.py

```diff
@@ -221,13 +268,58 @@ def _get_tags(self):
                 url = next_page["url"]
             else:
                 url = f'{self.registry_api}{next_page["url"]}'
-            response = self._request_get(url)
+            response = self._do_request(url)
 
             tags = response.json()['tags']
             all_tags.extend(tags)
 
         return all_tags
 
+    def _handle_cache_response(self, url):
+        # Uses a dispatch table to handle how different registries handle
+        # responses that are already present in the response cache. We will
+        # favor proper conditional requests if the registry supports it. The
+        # handling methods will update the response cache if needed.
+        return getattr(
+            self, self._HANDLE_RESPONSE_CACHE_METHODS[self.registry]
+        )(url)
+
+    def _handle_conditional_request(self, url):
+        # Handle response cache entries using conditional requests.
+        headers = {}
+        key = self._get_cache_key(url)
+        cached_response = self.response_cache[key]
+
+        etag = cached_response.headers.get("ETag")
+        if etag is not None:
+            headers["If-None-Match"] = etag
+
+        last_mod = cached_response.headers.get("Last-Modified")
+        if last_mod is not None:
+            headers["If-Modified-Since"] = last_mod
+
+        rsp = self._do_request(url, headers=headers)
+
+        if rsp.status_code != HTTPStatus.NOT_MODIFIED:
```

Comment:
hmmm, I see... you don't always cache. well, in this case, you should either return the `rsp` or the previously cached response:
```
if rsp.status_code == HTTPStatus.NOT_MODIFIED:
    cached_response
return rsp
```
so it plays nicelly with my previous suggestion :)

Reply:
Please notice that I invertedthe if logic, from `!=` to `==` to improve readability as well :)